src/legacy/ui/public/courier/fetch/components/shard_failure_table.tsx

```diff
@@ -0,0 +1,116 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import React, { useState, ReactElement } from 'react';
+// @ts-ignore
+import { EuiInMemoryTable, EuiButtonIcon } from '@elastic/eui';
+// @ts-ignore
+import { RIGHT_ALIGNMENT } from '@elastic/eui/lib/services';
+import { i18n } from '@kbn/i18n';
+import { ShardFailureDescription } from './shard_failure_description';
+import { ShardFailure } from './shard_failure_types';
+
+export interface ListItem extends ShardFailure {
+  id: string;
+}
+
+export function ShardFailureTable({ failures }: { failures: ShardFailure[] }) {
+  const itemList = failures.map((failure, idx) => ({ ...{ id: String(idx) }, ...failure }));
+  const initalMap = {} as Record<string, ReactElement>;
+
+  const [expandMap, setExpandMap] = useState(initalMap);
+
+  const collapseLabel = i18n.translate(
+    'common.ui.courier.fetch.shardsFailedModal.tableRowCollapse',
+    {
+      defaultMessage: 'Collapse',
+      description: 'Collapse a row of a table with failures',
+    }
+  );
+
+  const expandLabel = i18n.translate('common.ui.courier.fetch.shardsFailedModal.tableRowExpand', {
+    defaultMessage: 'Expand',
+    description: 'Expand a row of a table with failures',
+  });
+
+  const columns = [
+    {
+      align: RIGHT_ALIGNMENT,
+      width: '40px',
+      isExpander: true,
+      render: (item: ListItem) => (
+        <EuiButtonIcon
+          onClick={() => {
+            // toggle displaying the expanded view of the given list item
+            const map = Object.assign({}, expandMap);
+            if (map[item.id]) {
+              delete map[item.id];
+            } else {
+              map[item.id] = <ShardFailureDescription {...item} />;
+            }
+            setExpandMap(map);
+          }}
+          aria-label={expandMap[item.id] ? collapseLabel : expandLabel}
```

Comment:
Will there be more than one row rendered at a time? If so, could we change the labels to be something along the lines of `Expand {name of row, maybe index?} row` (or, if there isn't a concise way to name a row, `Expand row {i}`).

For repeated controls, we want to have unique names so users using a rotor (which cycles through all items of a type on a page) can find specific items (e.g., instead of hearing "Expand, button" 5 times, they hear a unique "Expand row {i}, button" and can more easily decide what to jump to). Let me know if that doesn't make sense or if you want a demo!

Reply:
that's a good point, I'll have a look

Reply:
I've refactored it to this state:

![Bildschirmfoto 2019-07-26 um 12 23 29](https://user-images.githubusercontent.com/463851/61945961-9ec7cb00-afa1-11e9-99c9-d220f7d90f4a.png)